src/clients/SpokePoolClient.ts

```diff
@@ -929,15 +904,16 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @note This method is used to find deposits that are outside of the search range of this client.
    */
   async findDeposit(depositId: number, destinationChainId: number, depositor: string): Promise<DepositWithBlock> {
-    // Binary search for block where SpokePool.numberOfDeposits incremented to fill.depositId + 1.
-    // This way we can get the blocks before and after the deposit with deposit ID = fill.depositId
-    // and use those blocks to optimize the search for that deposit. Stop searches after a maximum
-    // # of searches to limit number of eth_call requests. Make an eth_getLogs call on the remaining block range
-    // (i.e. the [low, high] remaining from the binary search) to find the target deposit ID.
+    // Binary search for block. This way we can get the blocks before and after the deposit with
+    // deposit ID = fill.depositId and use those blocks to optimize the search for that deposit.
+    // Stop searches after a maximum # of searches to limit number of eth_call requests. Make an
+    // eth_getLogs call on the remaining block range (i.e. the [low, high] remaining from the binary
+    // search) to find the target deposit ID.
+    //
     // @dev Limiting between 5-10 searches empirically performs best when there are ~300,000 deposits
     // for a spoke pool and we're looking for a deposit <5 days older than HEAD.
     const searchBounds = await this._getBlockRangeForDepositId(
-      depositId + 1,
```

Comment:
so assuming that your implementation of `_getBlockRangeForDepositId` is logically equivalent to the current implementation, why do you not  opt to search for `depositId + 1`? The reason we search for the +1 number currently is that we are interested ultimately in finding the block containing the deposit with `depositId`, which is emitted in the same block where `SpokePool.depositId` gets incremented +1. So by searching for the block where `SpokePool.depositId  = depositId + 1` we find the same block where the deposit with `depositId` is mined.

What do you think?

Reply:
`_getBlockRangeForDepositId` is now slightly modified in that we account for the deposit ID at a particular block as essentially `[depositCount[block-1], depositCount[block] - 1]`. As a result, the `+1` addition to account for this is no longer needed since we will get to the block that contains the deposit count without needing to modify the deposit we're looking for.

Reply:
@nicholaspai

@james-a-morris and I went through the old implementation and the new one around this specific question. It is logically equivalent, but the + 1 is taken care of inside the function. I thought this would probably be more intuitive to the caller who may not understand the precise way that deposit ids are incremented or how the contract variable is set. It seems more logical to ask "what was the block that this deposit was in" not "what block was the deposit number variable set to this value".